jigglypuffRL/classicalrl/sarsa/sarsa.py

```diff
@@ -0,0 +1,107 @@
+import gym
+import numpy as np
+
+class SARSA:
+    '''
+    State-Action-Reward-State-Action (SARSA)
+    Paper: http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.17.2539&rep=rep1&type=pdf
+    :param env: (Gym environment) The environment to learn from
+    :param max_iterations: (int) Maximum number of iterations per episode
+    :param max_epsiodes: (int) Maximum number of episodes
+    :param epsilon: (float) Epsilon for epsilon-greedy selection of the action
+    :param alpha: (float) Learning rate
+    :param gamma: (float) Discount Factor
+    :param lmbda: (float) Lambda for the eligibility traces
+    :param decay_rate_epsilon: (float) Decay rate for the epsilon of epsilon greedy action selection
+    :param thres_reward: (float) Threshold of the reward you want the training to stop
+    '''
+    def __init__(
+        self,
+        env,
+        max_iterations=1000,
+        max_episodes=1500001, 
+        epsilon=0.9,
+        alpha=0.1,
+        gamma=0.95,
+        lmbda=0.90,
+        decay_rate_epsilon=0.999,
+        thres_reward=0.8):
```

Comment:
On what basis are we giving `thres_reward` here? Won't it vary between environments?